src/main/kotlin/org/jetbrains/plugins/bsp/server/tasks/CollectProjectDetailsTask.kt

```diff
@@ -116,7 +153,19 @@ public class UpdateMagicMetaModelInTheBackgroundTask(
 
     private fun Sdk.isJdkNotAdded(): Boolean {
       val existingJdk = jdkTable.findJdk(this.name, this.sdkType.name)
-      return existingJdk == null || existingJdk.sdkAdditionalData !== this.sdkAdditionalData
+      return existingJdk == null || existingJdk.homePath != this.homePath
+    }
+
+    private fun addPythonSdk(pythonInfo: PythonBuildTarget) {
+      val allJdks = jdkTable.allJdks.toList()
+      val newSdk = SdkConfigurationUtil.createSdk(
+        allJdks,
+        pythonInfo.interpreter,
```

Comment:
```suggestion
        URI.create(pythonInfo.interpreter).toPath().toString(),
```

it should solve the issue